scripts/releases-history-to-gcs.py

```diff
@@ -57,50 +60,106 @@ def ignore_ssl_error(loop, context):
-        async with session.get("{}/releases/{}/revisions?limit=10000".format(balrog_api, r)) as resp:
-            return (await resp.json())["revisions"]
+def timeout_monkeypatch(storage, timeout_override):
+    orig_download_metadata = storage.download_metadata
+    orig_upload = storage.upload
+
+    async def download_metadata(self, *args, **kwargs):
+        kwargs["timeout"] = timeout_override
+        return await orig_download_metadata(self, *args, **kwargs)
+
+    async def upload(self, *args, **kwargs):
+        kwargs["timeout"] = timeout_override
+        return await orig_upload(self, *args, **kwargs)
+
+    storage.download_metadata = download_metadata
+    storage.upload = upload
+
+    return storage
+
 
+async def gather_gcloud_md5s(session, storage, bucket_name, loop):
+    md5s = {}
+    res = await storage.list_objects(bucket_name, session=session)
+    nextPageToken = res.get("nextPageToken")
+    for i in res.get("items"):
+        md5s[i["name"]] = i["md5Hash"]
+    while nextPageToken:
+        res = await storage.list_objects(bucket_name, session=session, params={"pageToken": nextPageToken})
+        nextPageToken = res.get("nextPageToken")
+        for i in res.get("items"):
+            md5s[i["name"]] = i["md5Hash"]
+    return md5s
 
-async def process_release(r, session, balrog_api, bucket, sem, loop):
+
+async def process_release(r, session, balrog_db, bucket, mysql_sem, gcs_sem, loop, gcloud_md5s):
     releases = defaultdict(int)
     uploads = defaultdict(lambda: defaultdict(int))
 
     print("Processing {}".format(r), flush=True)
-    for rev in await get_revisions(r, session, balrog_api, sem):
-        releases[r] += 1
-        async with sem:
-            old_version = await (await session.get("{}/history/view/release/{}/data".format(balrog_api, rev["change_id"]))).text()
-        old_version_hash = hashlib.md5(old_version.encode("ascii")).digest()
-        try:
-            current_blob = await bucket.get_blob("{}/{}-{}-{}.json".format(r, rev["data_version"], rev["timestamp"], rev["changed_by"]))
-            current_blob_hash = base64.b64decode(current_blob.md5Hash)
-        except aiohttp.ClientResponseError:
-            current_blob_hash = None
-        if old_version_hash != current_blob_hash:
-            blob = bucket.new_blob("{}/{}-{}-{}.json".format(r, rev["data_version"], rev["timestamp"], rev["changed_by"]))
-            await blob.upload(old_version, session)
-            uploads[r]["uploaded"] += 1
-        else:
-            uploads[r]["existing"] += 1
+    try:
+        async with mysql_sem:
+            for attempt in range(1, 6):
+                try:
+                    revisions = await loop.run_in_executor(
+                        None, balrog_db.execute, f"SELECT data_version, timestamp, changed_by, data FROM releases_history WHERE name='{r}'"
+                    )
+                    break
+                except:  # noqa: E722
+                    exc = traceback.format_exc()
```

Comment:
probably not worth fixing, but I'm pretty sure this would work as well:
```
except Exception:
   exc = traceback.format_exc()
   print(f"Caught exception while selection {r} history from db:\n{exc}")
```